hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenRejectHandler.java

```diff
@@ -126,7 +134,52 @@ public void handle(@NonNull final HandleContext context) throws HandleException
     @NonNull
     @Override
     public Fees calculateFees(final FeeContext feeContext) {
-        // Todo: Implement the logic for calculating fees
-        return Fees.FREE;
+        final var body = feeContext.body();
+        final var op = body.tokenRejectOrThrow();
+        final var config = feeContext.configuration();
+        final var tokenMultiplier =
+                requireNonNull(config).getConfigData(FeesConfig.class).tokenTransferUsageMultiplier();
+
+        final int totalRejections = op.rejections().size();
+        int numOfFungibleTokenRejections = 0;
+        int numOfNFTRejections = 0;
+        for (final var rejection : op.rejections()) {
+            if (rejection.hasFungibleToken()) {
+                numOfFungibleTokenRejections++;
+            } else {
+                numOfNFTRejections++;
+            }
+        }
+
+        final int weightedTokensInvolved = tokenMultiplier * totalRejections;
+        final int weightedFungibleTokens = tokenMultiplier * numOfFungibleTokenRejections;
+
+        final long bpt =
+                calculateBytesPerTransaction(weightedTokensInvolved, weightedFungibleTokens, numOfNFTRejections);
+        final long rbs = calculateRamByteSeconds(weightedTokensInvolved, weightedFungibleTokens, numOfNFTRejections);
+
+        return feeContext
+                .feeCalculator(getSubType(numOfNFTRejections))
+                .addBytesPerTransaction(bpt)
+                .addRamByteSeconds(rbs * USAGE_PROPERTIES.legacyReceiptStorageSecs())
```

Comment:
rbs is already multiplied with `USAGE_PROPERTIES.legacyReceiptStorageSecs()` in line 181. Need not do here again.

Reply:
Thats true. Great catch! I've adjusted it.